### PR TITLE
Add [Gen 9] Doubles Hackmons Cup

### PR DESCRIPTION
**Pull request by adrivrie**
Adds gen 9 doubles hackmons cup as a challenge-only format. Also moves CC2v2 from singles to randomized formats.

Discussed in the rands auth discord here: https://discord.com/channels/294651453279174656/1041098998624288990/1049303050042482748

### CODE REVIEW

File: config/formats.ts [hunk 1]
```diff
@@ -2089,6 +2081,24 @@ export const Formats: FormatList = [
...
+
+		mod: 'gen9',
+		gameType: 'doubles',
+		team: 'randomHC',
+		searchShow: false,
+		ruleset: ['HP Percentage Mod', 'Cancel Mod'],
+		banlist: ['Nonexistent'],
+	},
+	{
+		name: "[Gen 9] Challenge Cup 2v2",
```
Comment (AnnikaCodes) [inline config/formats.ts:2095]: Why move CC2v2 to Randomized Formats but not CC1v1?

Comment (adrivrie) [reply config/formats.ts:2095]: I figured the reason for having them in the "top slot" was to make a couple rands formats more visible while awaiting gen9, but it was a little grating to have a doubles format in the singles category, and we figured it'd be better in rands than doubles. I don't really mind either way however, just prefer cc2v2 not to be in singles.

Also, feel free to just close this as you already added DHC :)

**Comment by MathyFurret** [comment]
This is already done #9184

**Comment by adrivrie** [comment]
Indeed it is, thought it was already closed. Closing now!

**Pull request closed without merging**